docs/docs/sourcing-from-agilitycms.md

```diff
@@ -46,52 +50,274 @@ Install the dependencies:
+
+Each page in Agility is composed and generated dynamically at _build-time_ using a **masterPageTemplate** that you define in your **gatsby-config.js** plugin options.
+
+*gatsby-config.js*:
+
+```shell
+module.exports = {
+  siteMetadata: {
+    title: "Agility CMS Gatsby Starter",
+  },
+  plugins: [
+    ...
+    {
+      //the name of the plugin
+      resolve: "@agility/gatsby-source-agilitycms", 
+      //the options for our plugin
+      options: {
+        ...
+         //the page template that will be used to render Agility CMS pages
+        masterPageTemplate: "./src/AgilityPage.jsx"
+      },
+    },
+    ...
+  ]
+}
+```
+
+**AgilityPage.jsx:**
+
+```shell
+import React from "react"
+import { graphql } from "gatsby"
+...
+
+export const query = graphql`
+  query($pageID: Int!, $contentID: Int!, $languageCode: String!) {
+    agilitypage(languageCode: { eq: $languageCode }, itemID: { eq: $pageID }) {
+      pageJson
+    }
+    agilityitem(
+      languageCode: { eq: $languageCode }
+      itemID: { eq: $contentID }
+    ) {
+      itemJson
+    }
+  }
+`
+
+const AgilityPage = ({ pageContext, data }) => {
+  ...
+  return (
+    <>
+      <SEO
+        title={viewModel.page.title}
+        description={viewModel.page.seo.metaDescription}
+        keywords={viewModel.page.seo.metaKeywords}
+        ogImage={viewModel.dynamicPageItem?.customFields?.image?.url}
+      />
+      <PreviewBar isPreview={viewModel.isPreview} />
+      <div id="site-wrapper" className="flex flex-col min-h-screen">
+        <SiteHeader
+          languageCode={viewModel.languageCode}
+          isMultiLanguage={viewModel.isMultiLanguage}
+        />
+        <main className="flex-grow">
+          <AgilityPageTemplate {...viewModel} />
+        </main>
+        <SiteFooter />
+      </div>
+    </>
+  )
+}
+
+export default AgilityPage
+```
+
+[Learn more about How Agility Pages Work in Gatsby](https://help.agilitycms.com/hc/en-us/articles/360043035211)
+
+### How Page Modules Work
+
+Page Modules in Agility CMS are the functional components that make up a page. Editors use these to compose what type of content is on each page and in what order they appear. Developers define what page modules are available in the CMS and what fields they have. Each module defined in Agility CMS should correspond to a React Component in your Gatsby site.
+
+In the [agilitycms-gatsby-starter](https://github.com/agility/agilitycms-gatsby-starter) site, the name of the page module is used to find a corresponding react component that matches the same name. If a match is found, that component is dynamically imported and rendered.
+
+For example, if a module has a reference name of RichTextArea in the CMS, then while the page is being rendered by the gatsby-source-agilitycms plugin, it will look for `./src/components/agility-pageModules/RichTextArea.jsx` in the `./src/components/agility-pageModules` directory.
```

Comment:
```suggestion
For example, if a module has a reference name of RichTextArea in the CMS, then while the page is being rendered by the gatsby-source-agilitycms plugin, it will look for `src/components/agility-pageModules/RichTextArea.jsx` in the `src/components/agility-pageModules` directory.
```